exporter/collector/internal/normalization/benchmark_test.go

```diff
@@ -110,27 +91,171 @@ func BenchmarkNormalizeSummaryDataPoint(b *testing.B) {
+	normalizer.NormalizeNumberDataPoint(startPoint, id)
+
+	var ok bool
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		// each point decreases in value, which triggers a reset.
+		b.StopTimer()
+		newPoint := testNumberDataPoint()
+		newPoint.SetIntValue(int64(b.N - i))
+		b.StartTimer()
+		_, ok = normalizer.NormalizeNumberDataPoint(newPoint, id)
+		assert.True(b, ok)
+	}
+}
+
+func BenchmarkResetNormalizeHistogramDataPoint(b *testing.B) {
+	shutdown := make(chan struct{})
+	defer close(shutdown)
+	normalizer := NewStandardNormalizer(shutdown, zap.NewNop())
+	startPoint := testHistogramDataPoint()
+	startPoint.SetTimestamp(start)
+	startPoint.SetSum(float64(b.N + 1))
+	id := uint64(12345)
+	// ensure each run is the same by skipping the first call, which will populate caches
+	normalizer.NormalizeHistogramDataPoint(startPoint, id)
+
+	var ok bool
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		// each point decreases in value, which triggers a reset.
+		b.StopTimer()
+		newPoint := testHistogramDataPoint()
+		newPoint.SetSum(float64(b.N - i))
+		b.StartTimer()
+		_, ok = normalizer.NormalizeHistogramDataPoint(newPoint, id)
+		assert.True(b, ok)
+	}
+}
+
+func BenchmarkResetNormalizeExponentialHistogramDataPoint(b *testing.B) {
+	shutdown := make(chan struct{})
+	defer close(shutdown)
+	normalizer := NewStandardNormalizer(shutdown, zap.NewNop())
+	startPoint := testExponentialHistogramDataPoint()
+	startPoint.SetTimestamp(start)
+	startPoint.SetSum(float64(b.N + 1))
+	id := uint64(12345)
+	// ensure each run is the same by skipping the first call, which will populate caches
+	normalizer.NormalizeExponentialHistogramDataPoint(startPoint, id)
+
+	var ok bool
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		// each point decreases in value, which triggers a reset.
+		b.StopTimer()
+		newPoint := testExponentialHistogramDataPoint()
+		newPoint.SetSum(float64(b.N - i))
+		b.StartTimer()
+		_, ok = normalizer.NormalizeExponentialHistogramDataPoint(newPoint, id)
+		assert.True(b, ok)
+	}
+}
+
+func BenchmarkResetNormalizeSummaryDataPoint(b *testing.B) {
+	shutdown := make(chan struct{})
+	defer close(shutdown)
+	normalizer := NewStandardNormalizer(shutdown, zap.NewNop())
+	startPoint := testSummaryDataPoint()
+	startPoint.SetTimestamp(start)
+	startPoint.SetSum(float64(b.N + 1))
+	id := uint64(12345)
+	// ensure each run is the same by skipping the first call, which will populate caches
+	normalizer.NormalizeSummaryDataPoint(startPoint, id)
+
+	var ok bool
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		// each point decreases in value, which triggers a reset.
+		b.StopTimer()
+		newPoint := testSummaryDataPoint()
+		newPoint.SetSum(float64(b.N - i))
+		b.StartTimer()
+		_, ok = normalizer.NormalizeSummaryDataPoint(newPoint, id)
+		assert.True(b, ok)
+	}
+}
+
+var (
+	now   = pcommon.NewTimestampFromTime(time.Now())
+	start = pcommon.NewTimestampFromTime(time.Now().Add(-time.Minute))
+)
+
+func testNumberDataPoint() pmetric.NumberDataPoint {
```

Comment:
You mentioned that these functions are pretty slow relative to normalization which was throwing off the benchmarks. Was it a problem or are we OK with the shorter benchmark time workaround?

Reply:
it seemed to slightly increase the variability in CPU benchmarks.  Memory benchmarks were completely unchanged.  0.01 seconds is what the otel-go uses upstream, so it shouldn't be an issue.